handlers/misttriggers/broker.go

```diff
@@ -0,0 +1,80 @@
+package misttriggers
+
+import (
+	"context"
+	"sync"
+
+	"github.com/golang/glog"
+	"golang.org/x/sync/errgroup"
+)
+
+// Broker provides an interface for allowing multiple components of the app
+// to respond to Mist trigger events without putting all of the application
+// logic into the trigger handling code itself.
+
+// There are three different cases to account for on Mist triggers; and
+// accordingly the TriggerXXX functions have three different signatures:
+// 1. Purely informative, like STREAM_BUFFER. This can't be rejected back
+//    to Mist, so no error signature necessary. All of the callbacks can
+//    be fired in parallel.
+// 2. Allow/deny, like USER_NEW. We either let the new viewer come in
```

Comment:
I see that you didn't use `USER_NEW` (or `PUSH_REWRITE`) in the `Broker` right now. Is the intention to put them here later?

If yes, then I suggest creating some `// TODO: VID-xxx` to put them here later. Otherwise, it's a little confusing why you have it here in the comment, but not in the code.

Reply:
Yeah, I'd also vote for not making this PR bigger but for prioritising a follow-up to convert the others while this is still fresh in everyone's minds